### DISCUSSION TITLE
Why mantine doesn't use forwardRef?

### DISCUSSION THREAD

**Discussion by nghiepdev** [mantinedev/mantine#181] [Q&A] [answered]
Why `mantine` doesn't use `forwardRef` to wrap the component?

```tsx
<Input elementRef={ref} />
```
to
```tsx
<Input ref={ref} />
```

**Comment by rtivital** [comment, marked as answer]
Because we cannot support custom component pass through in TypeScript via `component` prop with `forwardRef`
